src/microbe_stage/systems/FluidCurrentsSystem.cs

```diff
@@ -116,7 +160,63 @@ protected override void Update(float delta, in Entity entity)
         var pos = new Vector2(position.Position.X, position.Position.Z);
         var vel = VelocityAt(pos) * Constants.MAX_FORCE_APPLIED_BY_CURRENTS;
 
-        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta;
+        float effectStrength = entity.Get<CurrentAffected>().EffectStrength;
+
+        if (effectStrength == 0)
+        {
+            effectStrength = 1;
+        }
+        else if (effectStrength == -1)
```

Comment:
```suggestion
        else if (effectStrength < 0)
```
I'll just commit this change so that there's no negative forces being applied as I think that could end up with really silly results.